mapper/pchain/tx_parser.go

```diff
@@ -0,0 +1,740 @@
+
+	utxoID := avax.UTXOID{TxID: txID, OutputIndex: outIndex}
+
+	// Do not add coin change during construction as txid is not yet generated
+	// and therefore UTXO ids would be incorrect
+	var coinChange *types.CoinChange
+	if !t.isConstruction {
+		coinChange = &types.CoinChange{
+			CoinIdentifier: &types.CoinIdentifier{Identifier: utxoID.String()},
+			CoinAction:     types.CoinCreated,
+		}
+	}
+
+	return &types.Operation{
+		Type: opType,
+		OperationIdentifier: &types.OperationIdentifier{
+			Index: int64(startIndex),
+		},
+		CoinChange: coinChange,
+		Status:     status,
+		Account:    &types.AccountIdentifier{Address: outAddrFormat},
+		Amount:     mapper.AtomicAvaxAmount(outBigAmount),
+		Metadata:   opMetadata,
+	}, nil
+}
+
+func (t *TxParser) isMultisig(utxoid avax.UTXOID) (bool, error) {
+	dependencyTx, ok := t.dependencyTxs[utxoid.TxID.String()]
+	if !ok {
+		return false, errFailedToCheckMultisig
+	}
+
+	utxoMap := getUTXOMap(dependencyTx)
+	utxo, ok := utxoMap[utxoid.OutputIndex]
+	if !ok {
+		return false, errFailedToCheckMultisig
+	}
+
+	addressable, ok := utxo.Out.(avax.Addressable)
+	if !ok {
+		return false, errFailedToCheckMultisig
+	}
+	isMultisig := len(addressable.Addresses()) != 1
+
+	return isMultisig, nil
+}
+
+func GetAccountsFromUTXOs(hrp string, dependencyTxs map[string]*DependencyTx) (map[string]*types.AccountIdentifier, error) {
+	addresses := make(map[string]*types.AccountIdentifier)
+	for _, dependencyTx := range dependencyTxs {
+		utxoMap := getUTXOMap(dependencyTx)
+
+		for _, utxo := range utxoMap {
+			addressable, ok := utxo.Out.(avax.Addressable)
+			if !ok {
+				return nil, errFailedToGetUTXOAddresses
+			}
+
+			addrs := addressable.Addresses()
+
+			if len(addrs) != 1 {
+				continue
+			}
+
+			addr, err := address.Format(mapper.PChainNetworkIdentifier, hrp, addrs[0])
+			addresses[utxo.UTXOID.String()] = &types.AccountIdentifier{Address: addr}
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return addresses, nil
+}
+
+func GetDependencyTxIDs(tx txs.UnsignedTx) ([]ids.ID, error) {
+	var txIds []ids.ID
+	switch unsignedTx := tx.(type) {
+	case *txs.ExportTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.ImportTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.AddValidatorTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.AddDelegatorTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.CreateSubnetTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.CreateChainTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.AddSubnetValidatorTx:
+		txIds = append(txIds, getUniqueTxIds(unsignedTx.Ins)...)
+	case *txs.RewardValidatorTx:
+		txIds = append(txIds, unsignedTx.TxID)
+	}
+
+	ids.SortIDs(txIds)
+
+	return txIds, nil
+}
+
+func getUniqueTxIds(ins []*avax.TransferableInput) []ids.ID {
+	txnIDs := make(map[string]ids.ID)
```

Comment:
Why don't we just keep the IDs in an `ids.Set`?

Reply:
Does ids.Set provide any benefits?